### PR TITLE
feat: add pylint

### PR DESCRIPTION
**Pull request by rjsparks**
This builds on #187, and should be evaluated after that is merged.

It brings in a copy of the datatracker's .pylintrc, which we should review carefully.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+# Specify a configuration file.
+#rcfile=
+
+# Python code to execute, usually for sys.path manipulation such as
+# pygtk.require().
+#init-hook=
+
+# Add files or directories to the blacklist. They should be base names, not
+# paths.
+ignore=CVS,migrations
```
Comment (rjsparks) [inline .pylintrc:12]: CVS is vestigial
I'm surprised we're ignoring migrations.

File: .pylintrc [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+# Add files or directories matching the regex patterns to the blacklist. The
```
Comment (rjsparks) [inline .pylintrc:14]: blacklist

File: .pylintrc [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+jobs=1
+
+# Allow loading of arbitrary C extensions. Extensions are imported into the
+# active Python interpreter and may run arbitrary code.
+unsafe-load-any-extension=no
+
+# A comma-separated list of package or module names from where C extensions may
+# be loaded. Extensions are loading into the active Python interpreter and may
+# run arbitrary code
+extension-pkg-whitelist=
```
Comment (rjsparks) [inline .pylintrc:35]: whitelist

We may not control this name, but I wonder if it's still a valid configuration name?

File: .pylintrc [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+
+# Allow optimization of some AST trees. This will activate a peephole AST
+# optimizer, which will apply various small optimizations. For instance, it can
+# be used to obtain the result of joining multiple strings with the addition
+# operator. Joining a lot of strings can lead to a maximum recursion error in
+# Pylint and this flag can prevent that. It has one side effect, the resulting
+# AST will be different than the one from reality. This option is deprecated
+# and it will be removed in Pylint 2.0.
+optimize-ast=no
```
Comment (rjsparks) [inline .pylintrc:44]: probably long gone - we are at pylint 3

File: .pylintrc [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+# 3 compatible code, which means that the block might have code that exists
+# only in one or another interpreter, leading to false positives when analysed.
+analyse-fallback-blocks=no
+
+
+[EXCEPTIONS]
+
+# Exceptions that will emit a warning when being caught. Defaults to
+# "Exception"
+overgeneral-exceptions=builtins.Exception
```
Comment (rjsparks) [inline .pylintrc:408]: anachronism?

File: .pylintrc [hunk 6]
```diff
@@ -0,0 +1,408 @@
+[MASTER]
```
Comment (rjsparks) [inline .pylintrc:1]: MASTER (new pylint generates MAIN)

File: .pylintrc (hunk 1 above)
Comment (rjsparks) [reply .pylintrc:12]: new pylint --generate-rcfile generates this same set

File: .pylintrc (hunk 3 above)
Comment (rjsparks) [reply .pylintrc:35]: from a newly generated rc:
```
# A comma-separated list of package or module names from where C extensions may
# be loaded. Extensions are loading into the active Python interpreter and may
# run arbitrary code. (This is an alternative name to extension-pkg-allow-list
# for backward compatibility.)
extension-pkg-whitelist=
```

File: .pylintrc (hunk 4 above)
Comment (rjsparks) [reply .pylintrc:44]: confirmed - no longer relevant

**Review by jennifer-richards — CHANGES_REQUESTED** [review]
I think we should take the lessons from here and go with #189 instead

**Pull request closed without merging**